wi-fi/lwip/main.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Phoenix-RTOS --- LwIP port
+ *
+ * LwIP Wi-Fi
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Ziemowit Leszczynski
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "lwip/sys.h"
+
+#include "whd_wifi_api.h"
+#include "cybsp.h"
+#include "cybsp_wifi.h"
+#include "cy_lwip.h"
+#include "cyabs_rtos.h"
+#include "cy_lwip_log.h"
+
+#include <string.h>
+#include <sys/minmax.h>
+#include <posix/utils.h>
+
+
+#define WIFI_THREAD_PRIO    4
+#define WIFI_THREAD_STACKSZ (4 * _PAGE_SIZE)
+
+#define WIFI_DEV_ID   0
+#define WIFI_DEV_NAME "/dev/wifi"
+
+#define AP_SSID          "DATARO"
+#define AP_SECURITY_MODE WHD_SECURITY_WPA2_AES_PSK
+#define AP_SECURITY_KEY  "12345678"
+#define AP_CHANNEL       1
+
+#define SNPRINTF_APPEND(fmt, ...) \
+	do { \
+		if (!overflow) { \
+			int n = snprintf(buf, size, fmt, ##__VA_ARGS__); \
```

Comment:
That's bad. This macro depends on `buf` and `size` variables being present...

Reply:
Copied from legacy code, don't blame for this!

Reply:
Yeah, I've seen other examples. That's sad, but not in scope of this PR, you're right